torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -141,6 +179,9 @@ def decompress(fut):
         q = fut.value()[0].div_(world_size)
```

Comment:
Comment on `torch.matmul(matrix.t(), p, out=q)`:
Maybe you can check if everyone still has the same `p` here.

Reply:
> `torch.manual_seed(rng.randint(1_000_000_000))`:

`with torch.random.fork_rng(devices=[]):` pass an empty list to `devices` arg, which means the RNG is not forked on any CUDA device.

Reply:
> A comment on `torch.matmul(matrix, q, out=p)`:
> Did you double-check that all workers have the same `q` here?

Yes

Reply:
> compute_q

Yes, p is used indeed, `p = fut.value()[0]` basically reads the updated value after allreduce(p).

Before` compute_q(fut)`, the following code runs:
1) P = MQ;
2) AllReduce(P)
```
    torch.matmul(matrix, q, out=p)
    allreduce_p_fut = dist.all_reduce(p, group=group_to_use, async_op=True).get_future()
```

Reply:
> Comment on `torch.matmul(matrix.t(), p, out=q)`:
> Maybe you can check if everyone still has the same `p` here.

Can verify that `p` is the same across processes, as it's the result of AllReduce.

Reply:
> `with torch.random.fork_rng(devices=[]):` pass an empty list to `devices` arg, which means the RNG is not forked on any CUDA device.

Yes, that makes sense. My comment was more about the `torch.manual_seed` which looks like it would seed all devices (even though they haven't been forked).